src/python/pants/engine/build_files.py

```diff
@@ -127,25 +115,12 @@ def collect_dependencies(item):
       else:
         maybe_append(key, value)
 
-  collect_dependencies(struct)
-
-  yield UnhydratedStruct(
-    next(build_address for build_address in addresses if build_address == address),
-    struct,
-    dependencies)
+  # Recursively collect inline dependencies from the fields of the struct into `inline_dependencies`.
+  collect_inline_dependencies(struct)
 
-
-@rule(TargetAdaptorContainer, [AddressMapper, UnhydratedStruct])
-def hydrate_struct(address_mapper, unhydrated_struct):
-  """Hydrates a Struct from an UnhydratedStruct and its satisfied embedded addressable deps.
-
-  Note that this relies on the guarantee that DependenciesNode provides dependencies in the
-  order they were requested.
-  """
-  dependencies = yield [Get(TargetAdaptorContainer, Address, a) for a in unhydrated_struct.dependencies]
+  # And then hydrate the inline dependencies.
+  dependencies = yield [Get(HydratedStruct, Address, a) for a in inline_dependencies]
   dependencies = [d.value for d in dependencies]
```

Comment:
I know this was like this before, but I find re-setting a variable to some derivative of itself, presumably of a different type, confusing. Can we rename one of these two uses of `dependencies` so that the transformation is clear?